src/aliceVision/lensCorrectionProfile/lcp.hpp

```diff
@@ -0,0 +1,725 @@
+        RadialDistortParam2 = 0.f;
+        RadialDistortParam3 = 0.f;
+        TangentialDistortParam1 = 0.f;
+        TangentialDistortParam2 = 0.f;
+        ScaleFactor = 1.f;
+        isEmpty = true;
+    }
+};
+
+struct PerspectiveModel
+{
+    int Version = -1;
+    float FocalLengthX = 0.f;
+    float FocalLengthY = 0.f;
+    float ImageXCenter = 0.5f;
+    float ImageYCenter = 0.5f;
+    float ResidualMeanError = 0.f;
+    float ResidualStandardDeviation = 0.f;
+    float RadialDistortParam1 = 0.f;
+    float RadialDistortParam2 = 0.f;
+    float RadialDistortParam3 = 0.f;
+    bool isEmpty = true;
+
+    void reset()
+    {
+        Version = -1;
+        FocalLengthX = 0.f;
+        FocalLengthY = 0.f;
+        ImageXCenter = 0.5f;
+        ImageYCenter = 0.5f;
+        ResidualMeanError = 0.f;
+        ResidualStandardDeviation = 0.f;
+        RadialDistortParam1 = 0.f;
+        RadialDistortParam2 = 0.f;
+        RadialDistortParam3 = 0.f;
+        isEmpty = true;
+    }
+};
+
+struct VignetteModel
+{
+    float FocalLengthX = 0.f;
+    float FocalLengthY = 0.f;
+    float ImageXCenter = 0.5f;
+    float ImageYCenter = 0.5f;
+    float VignetteModelParam1 = 0.f;
+    float VignetteModelParam2 = 0.f;
+    float VignetteModelParam3 = 0.f;
+    bool isEmpty = true;
+
+    void reset()
+    {
+        FocalLengthX = 0.f;
+        FocalLengthY = 0.f;
+        ImageXCenter = 0.5f;
+        ImageYCenter = 0.5f;
+        VignetteModelParam1 = 0.f;
+        VignetteModelParam2 = 0.f;
+        VignetteModelParam3 = 0.f;
+        isEmpty = true;
+    }
+};
+
+struct FisheyeModel
+{
+    int Version = -1;
+    float FocalLengthX = 0.f;
+    float FocalLengthY = 0.f;
+    float ImageXCenter = 0.5f;
+    float ImageYCenter = 0.5f;
+    float ResidualMeanError = 0.f;
+    float ResidualStandardDeviation = 0.f;
+    float RadialDistortParam1 = 0.f;
+    float RadialDistortParam2 = 0.f;
+    bool isEmpty = true;
+
+    void reset()
+    {
+        Version = -1;
+        FocalLengthX = 0.f;
+        FocalLengthY = 0.f;
+        ImageXCenter = 0.5f;
+        ImageYCenter = 0.5f;
+        ResidualMeanError = 0.f;
+        ResidualStandardDeviation = 0.f;
+        RadialDistortParam1 = 0.f;
+        RadialDistortParam2 = 0.f;
+        isEmpty = true;
+    }
+};
+
+/**
+ * @brief LensParam contains parameters of distortion, vignetting and chromatic aberation models
+ * for a set of camera settings (focal length, focus distance, aperture value).
+ * Detailed information on models can be found in the Adobe technical report
+ * "Adobe Camera Model" part of the documentation of the Adobe free tool Lens Profile Creator.
+ */
+class LensParam
+{
+public:
+    LensParam() = default;
+
+    /**
+    * @brief LensParam reset
+    */
+    void clear();
+
+    /**
+     * @brief Indicate that paramaters apply for a fisheye lens
+     * @return true if the fisheye model is the valid one
+     */
+    inline bool isFisheye() const { return _isFisheye; }
```

Comment:
There's no need to set `inline` for functions that are immediately defined within the class.